### PR TITLE
Array.Copy fixes

### LINKED ISSUE #10646: Array.Copy doesn't handle pointer-objref conversions
This should probably throw a `ArrayTypeMismatchException` like e.g. the `IntPtr[]` and `int*[]` case does, but it does something horrible instead:

```csharp
using System;

unsafe class Program
{
    static int Main()
    {
        int*[] inarr = new int*[] { (int*)1, (int*)2 };
        object[] oarr = new object[inarr.Length];
        Array.Copy(inarr, oarr, inarr.Length);
        Array.Copy(oarr, inarr, inarr.Length);
        return 100;
    }
}
```

Neither of these `Array.Copy` should succeed.

### PR DESCRIPTION
**Pull request by jkotas**
Fix #10646 and https://github.com/dotnet/corefx/issues/13816

### CODE REVIEW

**Comment by jkotas** [comment]
corefx test updates: https://github.com/dotnet/corefx/pull/19063

File: src/classlibnative/bcltype/arraynative.cpp [hunk 1]
```diff
@@ -435,10 +435,6 @@ ArrayNative::AssignArrayEnum ArrayNative::CanAssignArrayType(const BASEARRAYREF
     if (srcTH.IsInterface() && destElType != ELEMENT_TYPE_VALUETYPE)
         return AssignMustCast;
 
-    // Enum is stored as a primitive of type dest.
```
Comment (gkhanna79) [inline src/classlibnative/bcltype/arraynative.cpp]: Should we call these out for breaking change?

Comment (jkotas) [inline src/classlibnative/bcltype/arraynative.cpp]: I do not think that these are breaking changes.

**Comment by tarekgh** [comment]
LGTM. thanks for fixing it.

**Pull request merged**